### DISCUSSION TITLE
Expo Router: Type of href of the Link ?

### DISCUSSION THREAD

**Discussion by welschmoor** [expo/expo#27252] [Archived: Questions]
Hello, I am struggling to put a type on the href that I want passed through Props to a Link that I wrap.

```
import { Link } from 'expo-router'

type LinkWrapperProps = {
  href: ?????  // <- type of this
}

const LinkWrapper = ({ href }) => {

  return (
    <Link href={href} />
  )
}
```

Am I close? :
```
  href: LinkProps<ListLinkProps>['href']

  href:  LinkProps<"/">['href']
```

Typed routes, I of course want to preserve type safety, so I can't use string

**Comment by alexandrius** [comment]
```ts

import { type Href } from 'expo-router';

export interface TouchableScaleProps<T> extends PressableProps {
  activeScale?: number;
  disabled?: boolean;
  href?: Href<T>;
}

export interface LinkScaleProps<T> extends TouchableScaleProps<T> {
  href: TouchableScaleProps<T>['href'];
}

export function LinkScale<T>({ href, onPress, ...props }: LinkScaleProps<T>) {
  const resolvedHref = useMemo(() => {
    if (href == null) {
      throw new Error('Link: href is required');
    }
    return resolveHref(href);
  }, [href]);

  const linkProps = useLinkToPathProps({ href: resolvedHref });

  return (
    <TouchableScale
      {...props}
      onPress={(event) => {
        onPress?.(event);
        linkProps.onPress();
      }}
    />
  );
}
```

**Comment by aleganza** [comment]
```
import { type Href } from 'expo-router'

interface Props {
  href: Href<string>
}
```

**Reply by ierdss** [reply]
Thanks this is what I've been looking for.

**Reply by JClackett** [reply]
This doesn't account for dynamic urls, aka `/users/[id]`

Any idea how to type that?

**Reply by aleganza** [reply]
> This doesn't account for dynamic urls, aka `/users/[id]`
> 
> 
> 
> Any idea how to type that?

you should check into Href<T> and import the one for dynamic links I suppose

**Reply by JClackett** [reply]
Yeah the dynamic links type isn't exported for some reason

**Comment by Mookiies** [comment]
```ts
import { Link } from 'expo-router';
import { type ComponentProps } from 'react';

type Href = ComponentProps<typeof Link>['href'];
```

**Comment by hoangrank1** [comment]
import { Stack, Redirect } from 'expo-router';
import { type Href } from 'expo-router';

interface Props {
  href: Href<string>;
}

export default function AppLayout({ href = "/login" }: Props) {
  const isLoggedIn = true;

  if (!isLoggedIn) {
    return <Redirect href={href} />;
  }

  return <Stack screenOptions={{ headerShown: false }} />;
}
//////////////////////
still get the the error Type '"/login"' is not assignable to type 'Href<string>'. Can someone help me @@?
